sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -402,9 +413,30 @@ SessionPair startSession() {
       }
       previousSession = session;
 
-      session =
-          new Session(
-              options.getDistinctId(), user, options.getEnvironment(), options.getRelease());
+      String distinctId = null;
+
+      final User finalUser = getUser();
+      //      synchronized (userLock) {
```

Comment:
it's racey if we don't sync it. but as session is already locked, only a copy of User would be fine.

Reply:
Technically user would need to be readonly but yeah